### ISSUE TITLE
Suite finding doesn't work if the test discovery includes printed text

### ISSUE THREAD

**Issue by jakegt1** [kondratyev-nv/vscode-python-test-adapter#8]
If you have any sort of code that prints to stdout on import of a test file, it is read as a test suite and added to the tests (and subsequently breaks tests.)

**Comment by kondratyev-nv** [comment]
Thanks for reporting! Do you mind providing sample code to reproduce the issue?

**Comment by jakegt1** [comment]
I'm gonna take a guess and assume this will work:

```python
print("Hello, i am here to break tests")
from unittest import TestCase

class TestStuff(TestCase):

    def test_stuff():
        assert True
```

**Comment by jakegt1** [comment]
It's because when the code executes the python scripts in unittestScripts.ts, the output will include any output printed before the tests are run.

**Comment by darkvertex** [comment]
With support for pytest (#3) it might be worth ditching the custom discovery code altogether in favour of pytest's own functionality which will handle everything properly for both unittest and pytest tests (with or without output) and give you a consistent output that's not too hard to parse...

For example, let's make unittest files in both flavours. First we have `test_with_unittest_file.py` which will be:
```
import unittest

class MyTestCase(unittest.TestCase):

    def test_truthistrue(self):
        assert True == True

    def test_falseisfalse(self):
        assert False == False
```
and let's make another file called `test_with_pytest_file.py` with:
```
def test_truthistrue():
    assert True == True

def test_falseisfalse():
    assert False == False
```

Now let's see what pytest discovers with the `--collect-only` (discovery) mode:

```
> pytest --collect-only -s --cache-clear .
======================================== test session starts ========================================
platform linux2 -- Python 2.7.13, pytest-3.8.0, py-1.6.0, pluggy-0.7.1
rootdir: /mnt/users/alan/rdoenv/vscodemisc/dummytests, inifile:
plugins: cov-2.5.1
collected 4 items
<Module 'test_with_pytest_file.py'>
  <Function 'test_truthistrue'>
  <Function 'test_falseisfalse'>
<Module 'test_with_unittests_file.py'>
  <UnitTestCase 'MyTestCase'>
    <TestCaseFunction 'test_falseisfalse'>
    <TestCaseFunction 'test_truthistrue'>

=================================== no tests ran in 0.06 seconds ====================================
```
Look at that, it's all nicely laid out in pseudo-xml. lol

**Comment by jakegt1** [comment]
I initially thought this was due to having to parse through unittest discovery text, but actually it's not - the code runs the python function `unittest.TestLoader.discover`, which you can use to iterate through each test found via unittest then print out each module string, like so:

```
mypkg.test.unit.TestCase.test_stuff
mypkg.test.unit.TestCase.test_other
mypkg.test.functional.TestFunctionalCase.test_stuff
...
```

as so.

However, as the code *DOES* read what comes out of stdout from this function (it makes a subprocess and takes stdout as a string), if you have code that prints before the test is run, it'll be more like this:

```
I am printing something outside of the test definition
more printing
some other printing
mypkg.tests.unit.TestCase.test_stuff
...
```

I will note that iirc, pytest has the same issue.

**Comment by darkvertex** [comment]
> I will note that iirc, pytest has the same issue.

This is my bad, I used the wrong flags in my example.

I put a print in my file:
```
import unittest

print "Hello there."

class MyTestCase(unittest.TestCase):

    def test_truthistrue(self):
        assert True == True

    def test_falseisfalse(self):
        assert False == False
```

Then here is the proper command with `--capture=sys` to grab stdout and stderr from printing:

```
> pytest --collect-only --capture=sys --cache-clear
==================================================================================== test session starts ====================================================================================
platform linux2 -- Python 2.7.13, pytest-3.8.0, py-1.6.0, pluggy-0.7.1
rootdir: /mnt/users/alan/rdoenv/vscodemisc/dummytests, inifile:
plugins: cov-2.5.1
collected 2 items
<Module 'test_with_unittests_file_with_prints.py'>
  <UnitTestCase 'MyTestCase'>
    <TestCaseFunction 'test_falseisfalse'>
    <TestCaseFunction 'test_truthistrue'>
```

and the bad command I gave earlier:

```
> pytest --collect-only -s --cache-clear
==================================================================================== test session starts ====================================================================================
platform linux2 -- Python 2.7.13, pytest-3.8.0, py-1.6.0, pluggy-0.7.1
rootdir: /mnt/users/alan/rdoenv/vscodemisc/dummytests, inifile:
plugins: cov-2.5.1
collecting 0 items                                                                                                                                                                          Hello there.
collected 2 items
<Module 'test_with_unittests_file_with_prints.py'>
  <UnitTestCase 'MyTestCase'>
    <TestCaseFunction 'test_falseisfalse'>
    <TestCaseFunction 'test_truthistrue'>
```

**Comment by kondratyev-nv** [comment]
@jakegt1 I've merged your PR. However, I don't see `print("Hello, i am here to break tests")` breaking anything. I guess you had dots in print statement in your case and it did not break discovery, but some additional "suites" were added. Am I right?

**Comment by jakegt1** [comment]
Discovery was not broken but yes, additional "suites" were added that had names like the print statements that were made.

It may be that you have to have a different file, like so:

```python
# my_file.py
print("HELLO")
```

```python
# test_file.py
from unittest import TestCase
import my_file

class TestStuff(TestCase):

    def test_stuff():
        assert True
```

In the end i'm not too sure, but if anything i simply had print statements from imported files turning up in stdout, which the code attempted to parse through.
